src/app/button-page/button-page.component.spec.ts

```diff
@@ -166,4 +182,152 @@ describe('ButtonPageComponent', () => {
     expect(speechbarService.addButton).toHaveBeenCalled();
   });
 
+  it('should handle button click - do not speak on trigger', () => {
+    const button = { label: 'text', actions: [] };
+    component.handleButtonClick(<Button>button);
+    expect(speechbarService.sayButton).not.toHaveBeenCalled();
+  });
+});
+
+describe('ButtonPageComponent - speak on trigger', () => {
+  let component: ButtonPageComponent;
+  let fixture: ComponentFixture<ButtonPageComponent>;
+  let speechbarService: SpeechbarService;
+  let boardService: BoardService;
+  let configServiceStub: Partial<ConfigService>;
+
+  beforeEach(async(() => {
+    configServiceStub = {
+      buttonBehaviourConfig: {
+        triggerEvent: InteractionEventType.click,
+        speakOnTrigger: true
+      },
+      appearanceConfig: {
+        borderThickness: 2
+      },
+      scanningConfig: {
+        enabled: false,
+        time: 0
+      }
+    };
+
+    TestBed.configureTestingModule({
+      imports: [ MatRippleModule, HttpClientTestingModule ],
+      declarations: [ ButtonPageComponent, ObfButtonComponent, ProgressComponent, InteractionEventHandlerDirective ],
+      providers: [ BoardService, SpeechbarService, {provide: ConfigService, useValue: configServiceStub} ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    speechbarService = TestBed.get(SpeechbarService);
+    spyOn(speechbarService, 'clear');
```

Comment:
Might be worth adding a comment that these spies are to prevent actually calling the methods.  Just in case I try to delete them later thinking that they're not used!